plugins/destination/managed_writer.go

```diff
@@ -160,6 +160,10 @@ func (p *Plugin) writeManagedTableBatch(ctx context.Context, sourceSpec specs.So
 		if len(r.Data) < len(tables.Get(r.TableName).Columns) {
 			r.Data = append([]schema.CQType{sourceColumn, syncTimeColumn}, r.Data...)
 		}
+		colIndex := tables.Get(r.TableName).Columns.Index(schema.CqSyncTimeColumn.Name)
+		if p.spec.PartitionMinutes > 0 && colIndex != -1 {
+			_ = r.Data[colIndex].Set(r.Data[colIndex].Get().(time.Time).UTC().Round(time.Duration(p.spec.PartitionMinutes) * time.Minute))
```

Comment:
Can this get any longer? 😂  maybe make it into a helper fn? Repeated in plugin.go too

Reply:
Broke it into a standalone func